implementations/rust/ockam/ockam_api/src/lib.rs

```diff
@@ -469,3 +472,105 @@ impl<T: Encode<()>> ResponseBuilder<T> {
         Ok(())
     }
 }
+
+/// A newtype around `Cow<'_, str>` that borrows from input.
+///
+/// Contrary to `Cow<_, str>` the `Decode` impl for this type will always borrow
+/// from input so using it in types like `Option`, `Vec<_>` etc will not produce
+/// owned element values.
+#[derive(Debug, Clone, Encode, Decode, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cbor(transparent)]
+pub struct CowStr<'a>(#[b(0)] pub Cow<'a, str>);
```

Comment:
What about putting this to a separate file, or `ockam_core`, or even minicbor? And this is definitely separate commit